scipy/signal/signaltools.py

```diff
@@ -408,24 +401,13 @@ def fftconvolve(in1, in2, mode="full", axes=None):
     fslice = tuple([slice(sz) for sz in shape])
     # Pre-1.9 NumPy FFT routines are not threadsafe.  For older NumPys, make
     # sure we only call rfftn/irfftn from one thread at a time.
-    if not complex_result and (_rfft_mt_safe or _rfft_lock.acquire(False)):
-        try:
-            sp1 = np.fft.rfftn(in1, fshape, axes=axes)
-            sp2 = np.fft.rfftn(in2, fshape, axes=axes)
-            ret = np.fft.irfftn(sp1 * sp2, fshape, axes=axes)[fslice].copy()
-        finally:
-            if not _rfft_mt_safe:
-                _rfft_lock.release()
+    if not complex_result:
```

Comment:
@larsoner I mean these seemingly unrelated changes to `fftconvolve`.  Is this from not being rebased to the latest master?

Reply:
It's related in the sense that whatever triaging / complexity we have in `resample` should look a lot like what we have in `fftconvolve`. SciPy now requires a new enough `numpy` that we don't need to worry about this locking business anymore, so after this PR it shouldn't be in either code path (`fftconvolve` or `resample`).

Reply:
Those changes should be explained in the commit message.  [Was this project dragging on?](https://xkcd.com/1296/)

Reply:
Good call, done